Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/BoundsControl/Visuals/BoundsControlLinks.cs

```diff
@@ -155,31 +169,40 @@ private Vector3 GetLinkDimensions(Vector3 currentBoundsExtents)
             return (currentBoundsExtents * linkLengthAdjustor) + new Vector3(wireframeEdgeRadius, wireframeEdgeRadius, wireframeEdgeRadius);
         }
 
-        internal void Update(BoundsControlRotationHandles rotationHandles, Transform parent, Vector3 currentBoundsExtents)
+        internal void UpdateLinkPositions(ref Vector3[] boundsCorners)
         {
-            for (int i = 0; i < BoundsControlRotationHandles.NumEdges; ++i)
+            if (boundsCorners != null)
             {
-
-                if (links != null)
+                for (int i = 0; i < links.Count; ++i)
                 {
-                    links[i].position = rotationHandles.GetEdgeCenter(i);
+                    links[i].transform.position = BoundsControlVisualUtils.GetLinkPosition(i, ref boundsCorners);
+                }
+            }
+        }
 
+        internal void UpdateLinkScales(Vector3 currentBoundsExtents)
+        {
+            if (links != null)
+            {
+                for (int i = 0; i < links.Count; ++i)
+                {
+                    Transform parent = links[i].transform.parent;
                     Vector3 rootScale = parent.lossyScale;
                     Vector3 invRootScale = new Vector3(1.0f / rootScale[0], 1.0f / rootScale[1], 1.0f / rootScale[2]);
                     // Compute the local scale that produces the desired world space dimensions
                     Vector3 linkDimensions = Vector3.Scale(GetLinkDimensions(currentBoundsExtents), invRootScale);
 
-                    if (rotationHandles.GetAxisType(i) == CardinalAxisType.X)
+                    if (links[i].axisType == CardinalAxisType.X)
                     {
-                        links[i].localScale = new Vector3(wireframeEdgeRadius, linkDimensions.x, wireframeEdgeRadius);
+                        links[i].transform.localScale = new Vector3(wireframeEdgeRadius, linkDimensions.x, wireframeEdgeRadius);
                     }
-                    else if (rotationHandles.GetAxisType(i) == CardinalAxisType.Y)
+                    else if (links[i].axisType == CardinalAxisType.Y)
                     {
-                        links[i].localScale = new Vector3(wireframeEdgeRadius, linkDimensions.y, wireframeEdgeRadius);
+                        links[i].transform.localScale = new Vector3(wireframeEdgeRadius, linkDimensions.y, wireframeEdgeRadius);
```

Comment:
Suggestion:

Should the link class be smart enough to update itself?

Reply:
I was looking into that - the problem is that it needs the extents from the bounding box / bounds control. So we have to pass those in somehow. I can update the bounds in a different method and cache them and then have the update on it's own but that would just move the problem.
I was also thinking about calculating the extents in the class, but that would require further params from bounds control (flattenaxis) and if someone changes the bounds calculation in the control it might not match up anymore. So I don't really want to go that route. 
This applies to all other visual elements in bounds control as well. The extents are the only thing i couldn't really get rid of so far in an easy way.